frame/treasury/src/lib.rs

```diff
@@ -85,6 +87,9 @@ use frame_support::{
 pub use pallet::*;
 pub use weights::WeightInfo;
 
+const LOG_TARGET: &str = "runtime::treasury";
+
+pub type PayBalanceOf<T, I> = <<T as Config<I>>::Paymaster as Pay>::Balance;
```

Comment:
I think we should just make sure that the Pay implemented over `BalanceOf<T, I>`, and operate only with one balance type inside the pallet

Reply:
This sounds like a really bad idea, especially when we know that the balances pallet is actually a suboptimal solution, and can really be replaced by the assets pallet. There's no good reason why we aren't being more generic and general here.

Reply:
yes, agree, it's wrong.